package/MDAnalysis/coordinates/RDKit.py

```diff
@@ -95,10 +134,591 @@ def __init__(self, filename, **kwargs):
+    valence to get the Number of Unpaired Electrons (NUE).
+    If an atom has a negative NUE, it needs a positive formal charge (-NUE).
+    If two neighbouring atoms have UEs, the bond between them most
+    likely has to be increased by the value of the smallest NUE.
+    If after this process, an atom still has UEs, it's either a radical
+    (because one of its bonds was cut when creating the AtomGroup) or it needs
+    a negative formal charge of -NUE. Since these radical atoms can be detected
+    when looping over the bonds of the AtomGroup, only atoms that are not part
+    of this "terminal_atoms" list will be assigned a negative formal charge.
+
+    Parameters
+    ----------
+
+    mol : rdkit.Chem.rdchem.RWMol
+        The molecule is modified inplace and must have all hydrogens added
+
+    terminal_atom_indices : list
+        List of terminal atoms indices, i.e. atoms at the edges of a molecule
+    """
+
+    for atom in mol.GetAtoms():
+        # get NUE for each possible valence
+        expected_vs = PERIODIC_TABLE.GetValenceList(atom.GetAtomicNum())
+        current_v = atom.GetTotalValence() - atom.GetFormalCharge()
+        nue = [v - current_v for v in expected_vs]
+        # if there's only one possible valence state and the corresponding
+        # NUE is negative, it means we can only add a positive charge to
+        # the atom
+        if (len(nue) == 1) and (nue[0] < 0):
+            atom.SetFormalCharge(-nue[0])
+            mol.UpdatePropertyCache(strict=False)
+        # go to next atom if above case or atom has no unpaired electron
+        if (len(nue) == 1) and (nue[0] <= 0):
+            continue
+        else:
+            neighbors = atom.GetNeighbors()
+            # check if one of the neighbors has a common NUE
+            for i, na in enumerate(neighbors, start=1):
+                # get NUE for the neighbor
+                na_expected_vs = PERIODIC_TABLE.GetValenceList(
+                    na.GetAtomicNum())
+                na_current_v = na.GetTotalValence() - na.GetFormalCharge()
+                na_nue = [v - na_current_v for v in na_expected_vs]
+                # smallest common NUE
+                common_nue = min(
+                    min([i for i in nue if i >= 0], default=0),
+                    min([i for i in na_nue if i >= 0], default=0)
+                )
+                # a common NUE of 0 means we don't need to do anything
+                if common_nue != 0:
+                    # increase bond order
+                    bond = mol.GetBondBetweenAtoms(
+                        atom.GetIdx(), na.GetIdx())
+                    order = common_nue + 1
+                    bond.SetBondType(RDBONDORDER[order])
+                    mol.UpdatePropertyCache(strict=False)
+                    if i < len(neighbors):
+                        # recalculate nue for atom
+                        current_v = atom.GetTotalValence()
```

Comment:
I might be missing something, but is there a reason the formal charge gets used in the other NUE calculations, but not here?